s3/main.tf

```diff
@@ -1,6 +1,6 @@
 locals {
-  log_bucket_count = var.create_log_bucket ? 1 : 0
   log_bucket_name  = var.create_log_bucket ? "${var.bucket_name}-logs" : var.log_bucket_name
+  log_bucket_count = local.log_bucket_name ? 1 : 0
```

Comment:
Will this not always evaluate to `1`?

The default value for the variable `log_bucket_name` is an empty string, so `local.log_bucket_name` will always be present?

Might have got this wrong?

Reply:
oh, yep you are right! I will make that

`log_bucket_count = local.log_bucket_name != "" ? 1 : 0`

Reply:
have updated